pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -59,10 +63,68 @@ func (a *v1beta1PolicyApplier) JwtFilter(isXDSMarshalingToAnyEnabled bool) *http
 	return out
 }
 
+// All explaining code link can be removed before merging.
+func convertToIstioAuthnFilterConfig(jwtRules []*v1beta1.JWT) *authn_filter.FilterConfig {
+	p := authn_alpha.Policy{
+		// Targets are not referenced in proxy repo.
+		// https://github.com/istio/istio/blob/d8eb9d3699ef8945aa4cc6a1001a5455683df3b5/pilot/pkg/security/authn/v1alpha1/policy_applier.go#L265
+
+		// Origin will be optional since we don't reject req.
+		// Add Peers since we need that to trigger identity extraction in Authn filter.
```

Comment:
Why is peers needed for identity extraction? I thought just setting origin is enough.

Reply:
per offline discussion, the filter works as to require peer in order to populate `source.principal` from the x509 san uri.